python/orca/src/bigdl/orca/learn/pytorch/pytorch_ray_estimator.py

```diff
@@ -612,18 +612,30 @@ def shutdown(self, force=False):
 
         self.remote_workers = []
 
+    def _mean_reduce_stats(self, worker_stats, res_stats=None):
+        if not res_stats:
+            res_stats = {}
+        for stat_key, stat_value in worker_stats[0].items():
+            if isinstance(stat_value, numbers.Number): # loss
+                res_stats[stat_key] = np.nanmean(
+                    [s.get(stat_key, np.nan) for s in worker_stats])
+            elif isinstance(stat_value, torch.Tensor): # Accuracy
+                res_stats[stat_key] = torch.mean(
+                    torch.stack([stats[stat_key] for stats in worker_stats]))
+            elif isinstance(stat_value, dict): # profile
+                res_stats[stat_key] = self._mean_reduce_stats([stats[stat_key] for stats in worker_stats])
+            else:
+                res_stats[stat_key] = stat_value
+        return res_stats
+
     def _process_stats(self, worker_stats):
         stats = {
             "num_samples": sum(
                 stats.pop("num_samples", np.nan) for stats in worker_stats)
         }
 
-        for stat_key in worker_stats[0]:
-            if isinstance(worker_stats[0], numbers.Number):
```

Comment:
This is wrong, should get the value instead of isinstance on the dict.
Under such code, will always only use the stats of the first worker.
Also profile is not considered, which the value is a dict.